lightglue/disk.py

```diff
@@ -45,3 +55,15 @@ def forward(self, data: dict) -> dict:
             'keypoint_scores': scores.to(image),
             'descriptors': descriptors.to(image),
         }
+
+    def extract(self, img: torch.Tensor, **conf) -> dict:
```

Comment:
Share DISK.extract and SuperPoint.extract with a base Extractor class or by calling a shared util function `extrac(extractor, **conf)`.